src/oscar/apps/customer/forms.py

```diff
@@ -311,7 +311,9 @@ def __init__(self, user, *args, **kwargs):
                 self.fields[field_name].initial = getattr(user, field_name)
 
             # Ensure order of fields is email, user fields then profile fields
-            self.fields.keyOrder = user_field_names + profile_field_names
+            self.field_order = user_field_names + profile_field_names
```

Comment:
Could you clarify why we need to set `self.field_order` here? I feel like it's not necessary.

Reply:
Are you suggesting removing the line completely? Well, Django does have the field_order field in the BaseForm class and uses it for the obvious reasons (https://github.com/django/django/blob/456466d932830b096d39806e291fe23ec5ed38d5/django/forms/forms.py#L95). In my case, I was using a Profile class to add a language field along with the AUTH_PROFILE_MODULE setting and it crashed on that line. I think it's nice to keep the Django approach but the truth is , in my case, I don't need it. If anyone needs it, they can subclass UserAndProfileForm and implement it, true.

Reply:
I think you should set it to None, just like https://github.com/django/django/blob/456466d932830b096d39806e291fe23ec5ed38d5/django/forms/forms.py#L62 if it causes problems?